overlord/snapstate/snapmgr.go

```diff
@@ -564,7 +564,17 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	st.Unlock()
+	// XXX: this block is slightly ugly, find a pattern when we have more examples
 	err = m.backend.LinkSnap(newInfo)
+	if err != nil {
```

Comment:
no, we actually return the original LinkSnap error (as we did before, indeed there are tests that cover that occurence and notice that now we are trying to remove the partial work of LinkSnap)